enhancements/substitutes-for.md

```diff
@@ -0,0 +1,386 @@
+to be able to upgrade from 1.0.1-patched).
+
+This enhancement proposes we do so by allowing a bundle to specify which other bundles it "substitutes for".  This
+will effectively act as a search/replace in the existing bundles for any references to the bundles this bundle 
+substitutes for.  
+
+Continuing the example above, if 1.0.1-patch is annotated as substituting for 1.0.0, then the effective new upgrade 
+graph would be:
+
+1.0.0->1.0.1-patched->1.0.1
+
+
+The implications of this for opm add + the catalog db are that the substitutes-for metadata must be stored in the DB associated w/ each bundle.
+
+The expected flow is:
+
+1) opm add inserts the new bundle following all existing logic, but also adding the substitutes-for metadata
+2) Upon a list-bundles request, the operator-registry should return a substitution-corrected list of bundles based on substitutes replaces/skips references.
+3) In addition, the registry export command will need to perform substitutions on the CSVs as it exports them, to ensure their replaces/skips references are correct.
+
+
+Complications:
+
+1) the CSV itself contains other version references(beyond skips/replaces), but it should be ok not to perform updates on them because those references are not used by anything making upgrade decisions
+
+2) skipRange logic could theoretically get invalidated (1.2 substitutes for 1.1, but some bundle's skipRange is "skipRange: <1.2").  Since this feature is primarily being added for CVE rebuilds which we expect to only bump the patch release, this should simply be documented as a known limitation/caveat around using substitutes-for (that it does not allow substitution consideration during skipRange upgrades)
+
+3) If C substitutes B and B substituted A, it should be assumed that C substitutes A.  The db metadata and substitution processing needs to reflect this.
+
+
+## Motivation
+
+The primary motivation of this feature is to provide a way for the tooling to automatically rebuild an 
+operator bundle in response to CVEs in the operator's dependencies and then add the new bundle to the 
+operator catalog in such a way that clusters can upgrade to the new bundle from the unpatched bundle, 
+as well as upgrade from the patched bundle to newer bundles.
+
+### Goals
+
+* Patched bundles can be published that result in an intact upgrade graph, without having to modify
+other previously published bundles. (e.g. users do not end up stranded because they upgrade to a rebuilt
+bundle which cannot be upgraded to the head of another channel).
+* Future bundles that reference the unpatched version of a bundle, when a patched version exists in the
+catalog, have their references updated to include the patched versions.
+* Clusters running a version of the bundle that predates the vulnerable bundle are not forced to
+upgrade into the vulnerable bundle before being able to upgrade to the patched bundle.
```

Comment:
this sounds like what `skips` should have done initially.

just thinking out loud: the primary reason we can't just update skips to do this is because some folks explicitly install skipped bundles today with Subscriptions (only possible in OCP < 4.6)